pkg/controllers/default-resources.go

```diff
@@ -203,14 +210,14 @@ func scafoldAWSNodePoolPlatform(infraOut *aws.CreateInfraOutput) *hyp.AWSNodePoo
 	}
 }
 
-func ScaffoldNodePool(namespace string, infraId string, np *hypdeployment.HypershiftNodePools) *hyp.NodePool {
+func ScaffoldNodePool(hyd *hypdeployment.HypershiftDeployment, np *hypdeployment.HypershiftNodePools) *hyp.NodePool {
```

Comment:
@jnpacker does this refactor make sense to you, or would it break any logic between hypershiftDeployment and its nodepools?

My local env is broken... so have to run this with you.

Reply:
makes sense.